module/programs/helper.js

```diff
@@ -1057,19 +1057,31 @@ module.exports = class ProgramsHelper {
                 status: httpStatusCode.bad_request.status
               }
             }
+            
           }
-          pushProgramUsersDetailsToKafka = true; 
+          
+        }
+
+        // if requestForPIIConsent Is false and user not joined program till now then set pushProgramUsersDetailsToKafka = true; 
+        // if requestForPIIConsent == true and data.consentShared value is true which means user interacted with the consent popup set pushProgramUsersDetailsToKafka = true;
+        if((programData[0].hasOwnProperty('requestForPIIConsent') && programData[0].requestForPIIConsent === false && !programUsersDetails.length > 0) ||
+          ((programData[0].hasOwnProperty('requestForPIIConsent') && programData[0].requestForPIIConsent === true) && (data.hasOwnProperty('consentShared') && data.consentShared == true) ) ){
+            pushProgramUsersDetailsToKafka = true; 
         }
         
         //create or update query
         const query = { 
           programId: programId,
           userId: userId
         };
-        
+        //if a resource is started
```

Comment:
@VISHNUDAS-tunerlabs Can this both if condition be removed and values set in programUsersData on Line 1018

Reply:
@aks30 no. Line number 1018 will only happen if user didn't join program before, but some times both these value we may get in second calls(once the join already happened). Ex: user join program from program tile. Then start a resource Also front end calls join api with  consentShared=false first time then consentShared=true second time